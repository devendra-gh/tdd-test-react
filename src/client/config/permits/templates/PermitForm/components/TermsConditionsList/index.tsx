import React from 'react';
import { IVariables } from '@tamm/app-composer';
import Form from '@tamm/ui-lib-v2-form';
import Checkbox from '@tamm/ui-lib-v2-checkbox';
import './TermsConditionsList.less';

/**
 * TermsConditionsList component
 * @param {Object} props
 * @returns {JSX}
 */
function TermsConditionsList(props: IVariables) {
  const {
    inputGroups,
    handleChange,
    values,
    i18n,
    startShowingErrors,
    name: stateKey,
  } = props;
  const { [stateKey]: termsConditions } = values;

  const handleToggleCheckbox = (fieldName: string) => {
    return () => {
      handleChange({
        [stateKey]: {
          ...termsConditions,
          [fieldName]: !termsConditions[fieldName],
        },
      });
    };
  };

  return (
    inputGroups && (
      <Form name={props.name} id={props.name}>
        {inputGroups.map((formGroups: IVariables) => (
          <Form.Fieldset key={`${formGroups.groupKey}.${formGroups.stateKey}`}>
            <div className="marginT40">
              {formGroups.title && (
                <p className="termsTitle">{i18n(formGroups.title)}</p>
              )}
              {formGroups.conditions && formGroups.conditions.length > 0 && (
                <ul className="PermitForm__form-tac-ui">
                  {formGroups.conditions.map(
                    (condition: string, key: number) => (
                      <li key={String(key)}>{i18n(condition)}</li>
                    ),
                  )}
                </ul>
              )}
            </div>
            {formGroups.fields.map((fileConfig: IVariables) => {
              const {
                name,
                label,
                validationConfig,
                ...restProps
              } = fileConfig;

              return (
                <React.Fragment key={`${name}.${formGroups.groupKey}`}>
                  <Checkbox
                    {...restProps}
                    checked={termsConditions[name]}
                    label={label}
                    name={name}
                    onChange={handleToggleCheckbox(name)}
                    validateStatus={
                      startShowingErrors &&
                      validationConfig &&
                      !termsConditions[name]
                        ? 'error'
                        : ''
                    }
                  />
                </React.Fragment>
              );
            })}
          </Form.Fieldset>
        ))}
      </Form>
    )
  );
}

export default TermsConditionsList;
