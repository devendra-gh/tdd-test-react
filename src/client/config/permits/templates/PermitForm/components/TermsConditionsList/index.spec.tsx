import React from 'react';
import { shallow } from 'enzyme';
import TermsConditionsList from './index';
// import { checkValidationField } from 'client/config/utils/checkValidation';

// jest.mock('client/config/utils/checkValidation', () => ({
//   checkValidationField: {
//     REQUIRED: 'REQUIRED'
//   }
// }));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('Permit Type form', () => {
  let props: any;

  beforeEach(() => {
    props = {
      i18n: jest.fn(),
      handleToggleCheckbox: jest.fn(),
      values: {
        termsConditions: {
          approved: true,
        },
      },
      inputGroups: [
        {
          name: 'test1',
          title: 'test',
          groupKey: 'Gkey',
          stateKey: 'Skey',
          conditions: ['term1'],
          fields: [
            {
              validationConfig: {
                type: 'REQUIRED',
              },
              elementType: 'checkBox',
              name: 'approved',
              label: 'fieldLalel',
              // onChange: () => jest.fn(),
            },
          ],
        },
      ],
      handleChange: jest.fn(),
      startShowingErrors: true,
      name: 'termsConditions',
    };
  });

  it('should successfully render when serviceType is provided', () => {
    const component = shallow(<TermsConditionsList {...props} />);
    expect(component).toMatchSnapshot();
  });
  // eslint-disable-next-line no-console
  console.log = () => {};
  console.info = () => {};
  console.warn = () => {};
  console.error = () => {};

  // describe('checkbox', () => {
  //   it('should handle checkbox change event with termsConditions.approved as false', () => {
  //     props.values.termsConditions.approved = false;
  //     const component = shallow(<TermsConditionsList {...props} />);
  //     const Checkbox = component.find('Memo(Checkbox)');
  //     const allProps: any = Checkbox.props();
  //     const { onChange } = allProps;
  //     onChange('approved');
  //     expect(props.handleToggleCheckbox).toBeTruthy();
  //   });
  // });
});
