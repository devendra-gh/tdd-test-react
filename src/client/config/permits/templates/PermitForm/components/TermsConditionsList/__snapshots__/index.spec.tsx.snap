// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permit Type form should successfully render when serviceType is provided 1`] = `
<LocalizedComponent(Form)
  backButton={null}
  cancelLink={null}
  help=""
  i18n={[Function]}
  id="termsConditions"
  name="termsConditions"
  submitButton={null}
  validateStatus={null}
>
  <Form.Fieldset
    buttonGroup={false}
    gapSize="default"
    key="Gkey.Skey"
    title=""
    twoColumns={false}
  >
    <div
      className="marginT40"
    >
      <p
        className="termsTitle"
      />
      <ul
        className="PermitForm__form-tac-ui"
      >
        <li
          key="0"
        />
      </ul>
    </div>
    <Checkbox
      checked={true}
      elementType="checkBox"
      label="fieldLalel"
      name="approved"
      onChange={[Function]}
      validateStatus=""
    />
  </Form.Fieldset>
</LocalizedComponent(Form)>
`;
